docs/sources/clients/promtail/scraping.md

```diff
@@ -154,6 +154,28 @@ Keep in mind that labels prefixed with `__` will be dropped, so relabeling is re
     target_label: syslog_identifier
 ```
 
+## Gcplog scraping
+Promtail supports scraping cloud resource logs(say GCS bucket logs, Load Balancer logs, Kubernetes Cluster logs) from GCP.
+Configs are set in `gcplog` section in `scrape_config`
+
+```yaml
+  - job_name: gcplog
+    gcplog:
+      project_id: "my-gcp-project"
+      subscription: "my-pubsub-subscription"
+      use_incoming_timestamp: false # default rewrite timestamps.
+      labels:
+        job: "gcplog"
+```
+Here `project_id` and `subscription` are the only required fields.
+
+- `project_id` is the GCP project id.
+- `subscription` is the GCP pubsub subscription where promtail can consume log entries from.
+
+Before using `gcplog` target, GCP should be configured with pubsub subscription to receive logs from.
```

Comment:
Could you link a GCP documentation on how do to that. It is also worth adding information about how credentials are discovered and what roles are required.

Reply:
well. they do have some docs, but those are very generic and also not a single doc with all the steps that we need. 

What we actually need is our own doc explaining how we provision the cloud to extract logs from cloud resources. Combination of Logging Sink + Pubsub configs

Just thought of writing it on another PR and link it with this doc. wdyt?

Reply:
Sure another PR will do, it just lack of information on how to get started.